client/client.go

```diff
@@ -2432,6 +2427,26 @@ DISCOLOOP:
 
 	consulLogger.Info("discovered following servers", "servers", nomadServers)
 
+	// Check if the list of servers discovered is identical to the list we already have
```

Comment:
Hm, I think this code would be better placed in `servers.SetServers`. That way it could be done inside the server manager's lock and prevent concurrent RPCs from rotating the list and making this comparison stale.

Not a big deal as there's no data race here, just the possibility of a comparison against stale data which shouldn't do much harm.

Reply:
(It would be easily unit testable in server manager as well)

Reply:
The method for comparing whether the two server lists are equal in `servers.SetServers` does a sort - https://github.com/hashicorp/nomad/blob/9da25adc548acaf0443ce4c9f084d053ae496ddc/client/servers/manager.go#L209 
This approach avoids that sort step/modification. I can move over the code though and change the logic in SerServers to avoid the sort of existing servers, which modifies the observable server list.